Lars-02/Smoelenboek
# Mail Setup with Mailtrap

# Implement Password Email View

## Description
Sends mail to mailtrap through filled in address, if it exists in the database.

## Testing

For testing, pull in this branch and follow the following steps:
0. Make sure your env file is up to date with the changes in the env.example file.
1. Sign in on mailtrap
   Email: julian.afrika@gmail.com
   Password: smoelenboek
2. In phpstorm, click on "Database" at the top right of the screen.
3. Click on the "+" symbol.
4. Select data source "MariaDB".
5. Fill in "root" as username. (If the application wants you to authorise yourself via popup, just fill in root and click ok)
3. Select "apply" and then ok.
4. Run php artisan migrate:fresh in terminal.
5. You should now see smoelenboek@localhost with a schema in the database window.
6. In this schema, double-click users and click on the "+" above the users table.
7. Fill in an email and hashed password. (Use online generator to hash password, for instance: https://passwordsgenerator.net/md5-hash-generator/ )
8. Click on submit (green arrow pointing at "DB").
9. Run php artisan serve and navigate to the forgot password page. (http://*localhost IP*/password/reset)
10. Fill in email and click on "wachtwoord resetten".
11. Now go to the mailtrap inbox and select the newly received mail. In this mail there should be a link to the reset password page.

## Changes

The only changes are in .env.example. Here, the MAIL details were edited to support mailtrap. Also ran "php artisan vendor:publish --tag=laravel-mail" to import and edit style folders for mail.

## Screenshots

### Received Email
![image](https://user-images.githubusercontent.com/63709758/111305405-338baa80-8657-11eb-9afb-7ddbb136190a.png)
